test/external/pendle/PendleYtGLP2024IsolationModeTokenVaultV1.ts

```diff
@@ -0,0 +1,332 @@
+    const ytBal = await underlyingToken.balanceOf(core.hhUser1.address);
+    await vault.depositIntoVaultForDolomiteMargin(defaultAccountNumber, ytBal);
+
+    snapshotId = await snapshot();
+  });
+
+  beforeEach(async () => {
+    snapshotId = await revertToSnapshotAndCapture(snapshotId);
+  });
+
+  describe('#isExternalRedemptionPaused', () => {
+    it('should work normally', async () => {
+      expect(await vault.isExternalRedemptionPaused()).to.be.false;
+    });
+
+    it('should work when owner paused syGLP', async () => {
+      expect(await vault.isExternalRedemptionPaused()).to.be.false;
+      const syGlp = IPendleSyToken__factory.connect(await pendleRegistry.syGlpToken(), core.hhUser1);
+      const owner = await impersonate(await syGlp.owner(), true);
+      await syGlp.connect(owner).pause();
+      expect(await vault.isExternalRedemptionPaused()).to.be.true;
+    });
+  });
+
+  describe('#redeemDueInterestAndRewards', () => {
+    it('should work normally', async () => {
+      expectWalletBalance(vault.address, core.tokens.weth, ZERO_BI);
+      expectWalletBalance(core.hhUser1.address, core.tokens.weth, ZERO_BI);
+
+      await increaseToTimestamp((await underlyingToken.expiry()).toNumber());
+      const rewardDeposit = [{ marketId: core.marketIds.weth, depositIntoDolomite: true }];
+      await vault.connect(core.hhUser1).redeemDueInterestAndRewards(true, true, rewardDeposit);
+
+      const account = { owner: core.hhUser1.address, number: defaultAccountNumber };
+      const balance = await core.dolomiteMargin.getAccountWei(account, core.marketIds.weth);
+      expect(balance.sign).to.eq(true);
+      expect(balance.value).to.be.gt(ZERO_BI);
+      expectWalletBalance(vault.address, core.tokens.weth, ZERO_BI);
+      expectWalletBalance(core.hhUser1.address, core.tokens.weth, ZERO_BI);
+    });
+
+    it('should send rewards to user', async () => {
+      expectWalletBalance(vault.address, core.tokens.weth, ZERO_BI);
+      expectWalletBalance(core.hhUser1.address, core.tokens.weth, ZERO_BI);
+
+      await increaseToTimestamp((await underlyingToken.expiry()).toNumber());
+      const rewardDeposit = [{ marketId: core.marketIds.weth, depositIntoDolomite: false }];
+      await vault.connect(core.hhUser1).redeemDueInterestAndRewards(true, true, rewardDeposit);
+
+      expect(await core.tokens.weth.balanceOf(core.hhUser1.address)).to.be.gt(ZERO_BI);
+      expectWalletBalance(vault.address, core.tokens.weth, ZERO_BI);
+    });
+
+    it('should fail when not called by vault owner', async () => {
+      const rewardDeposit = [{ marketId: core.marketIds.weth, depositIntoDolomite: false }];
+      await expectThrow(
+        vault.connect(core.hhUser2).redeemDueInterestAndRewards(true, true, rewardDeposit),
+        `IsolationModeTokenVaultV1: Only owner can call <${core.hhUser2.address.toLowerCase()}>`
+      );
+    });
+  });
+
+  describe('#transferIntoPositionWithOtherToken', () => {
```

Comment:
Rename to `transferFromPositionWithOtherToken`

Be sure to add the new test cases for the added logic discussed in the corresponding Solidity file